discovery/targetgroup/targetgroup.go

```diff
@@ -77,9 +76,7 @@ func (tg *Group) UnmarshalJSON(b []byte) error {
 		Labels  model.LabelSet `json:"labels"`
 	}{}
 
-	dec := json.NewDecoder(bytes.NewReader(b))
-	dec.DisallowUnknownFields()
-	if err := dec.Decode(&g); err != nil {
+	if err := json.Unmarshal(b, &g); err != nil {
```

Comment:
I'm removing the previous strict unmarshaling here (for detecting unknown fields), because `json.Unmarshal()` does not support it. The question is whether that is ok. It seems in all the other `json.Unmarshal()` cases, it's also ok for us.

Reply:
json unmarshal is strict.

Reply:
https://golang.org/pkg/encoding/json/#Unmarshal

"By default, object keys which don't have a corresponding struct field are ignored (see Decoder.DisallowUnknownFields for an alternative)."